src/main/java/nextstep/subway/applicaion/dto/FavoriteCreateRequest.java

```diff
@@ -0,0 +1,31 @@
+package nextstep.subway.applicaion.dto;
+
+import lombok.Getter;
+import nextstep.member.domain.Member;
+import nextstep.subway.domain.Favorite;
+import nextstep.subway.domain.Station;
+
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+import java.util.function.Function;
+
+@Getter
+public class FavoriteCreateRequest {
+
+    @NotNull(message = "source is null")
+    private Long source;
+
+    @NotNull(message = "target is null")
+    private Long target;
+
+    public Favorite toEntity(Function<Long, Optional<Station>> findFunction, Member member) {
+        return Favorite.of(
+                member,
+                findStation(source, findFunction),
+                findStation(target, findFunction));
+    }
```

Comment:
```suggestion
    public Favorite toEntity(Function<Long, Station> func, Member member) {
        return Favorite.of(
                member,
                func.apply(source),
                func.apply(target));
    }
```

`StationService`에 구현되어있는 `findById` 메서드를 이용하면 이렇게 구현도 가능할 것 같아요 👍